tests/functional/RemoteWebDriverTest.php

```diff
@@ -245,4 +245,27 @@ public function testShouldSaveScreenshotToFile()
 
         unlink($screenshotPath);
     }
+
+    /**
+     * @covers ::takeScreenshot
+     */
+    public function testShouldSaveScreenshotToFileInSubDirectory()
+    {
+        if (!extension_loaded('gd')) {
+            $this->markTestSkipped('GD extension must be enabled');
+        }
+        if ($this->desiredCapabilities->getBrowserName() == WebDriverBrowserType::HTMLUNIT) {
+            $this->markTestSkipped('Screenshots are not supported by HtmlUnit browser');
+        }
+
+        $screenshotPath = sys_get_temp_dir() . '/fb-webdriver/dir/selenium-screenshot.png';
+
+        $this->driver->get($this->getTestPageUrl('index.html'));
+
+        $this->driver->takeScreenshot($screenshotPath);
+
+        $this->assertFileExists($screenshotPath);
+
+        unlink($screenshotPath);
```

Comment:
This does not remove the directory after the test, only the file. This should help:

```php
rmdir(dirname($screenshotPath));
```